dwave/cloud/coders.py

```diff
@@ -237,4 +242,59 @@ def decode_qp_numpy(msg, return_matrix=True):
         result['active_variables'] = result['active_variables'].tolist()
         result['solutions'] = result['solutions'].tolist()
 
+    # include problem type
+    if 'type' in msg:
+        result['problem_type'] = msg['type']
+
+    return result
+
+
+def encode_problem_as_bq(problem, compress=False):
+    """Encode the binary quadratic problem for submission the `bq` data format.
+
+    Args:
+        problem (:class:`~dimod.BinaryQuadraticModel`):
+            Binary quadratic model.
+
+    Returns:
+        encoded submission dictionary
```

Comment:
While I'm there anyway: "Encoded submission dict."

Reply:
@JoelPasvolsky, feel free to make a(nother) pass through the cloud client docs. I refactored docs in functions that I touched in this PR, but most of the documentation hasn't been updated since the times of microclient.